Remove older doc folder

csubira opened the pull request:
As part of **Update source doc branches and config file with new origin and versions/tags**:[CartoDB/developers#537](https://github.com/CartoDB/developers/issues/537)

VictorVelarde commented:
Hi @csubira I have found some mismatch between the current guides in the branch and the ones currently published.

![image](https://user-images.githubusercontent.com/458196/54040659-39c43200-41c6-11e9-86a8-f8b8d426e5bb.png)

These are the missing ones:
- [ ] **Clustering**: missing. And the main title in production has a broken style: `s## Clustering`
- [ ] **Ground overlays**: missing. And the first image doesn't display at production
- [ ] **Pro extensions**: a *proextensions.md* doc was deleted in this branch. Unless we have an explicit issue on this, I guess we would have to recover it.
- [ ] **Performance enhancements**: missing.

csubira commented:
Yes! In fact, even if there was a documentation folder in master, the dev center was fetching the info from the branch `doc-refactor-tabs`. So, I updated this branch with the content missing from that branch (as @VictorVelarde listed above) and also changed the older links to deprecated docs site. Now testing if everything shows properly in [staging](http://537-update-default-branches.developers.carto-staging.com/developers/mobile-sdk) before merging.
